tests/frontier/precompiles/test_precompile_absence.py

```diff
@@ -0,0 +1,72 @@
+"""
+abstract: Test Calling Precompile Range (close to zero)
+
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import Account, Address, Alloc, Bytecode
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_tools import StateTestFiller, Storage, Transaction
+
+UPPER_BOUND = 0x101
+RETURNDATASIZE_OFFSET = 0x10000000000000000  # Must be greater than UPPER_BOUND
+
+
+@pytest.mark.parametrize(
+    "calldata_size",
+    [
+        pytest.param(0, id="empty_calldata"),
+        pytest.param(31, id="31_bytes"),
+        pytest.param(32, id="32_bytes"),
+    ],
+)
+@pytest.mark.valid_from("Byzantium")
+def test_precompile_absence(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    calldata_size: int,
+):
+    """
+    Test that addresses close to zero are not precompiles unless active in the fork.
+    """
+    active_precompiles = fork.precompiles()
+    storage = Storage()
+    call_code = Bytecode()
+    for address in range(1, UPPER_BOUND + 1):
+        if Address(address) in active_precompiles:
+            continue
+        call_code += Op.SSTORE(
+            address,
+            Op.CALL(address=address, args_size=calldata_size),
```

Comment:
So the call is 1 in either case. Is it?

Also, need to check that precompiles before are also precompiles

Reply:
The idea is that the call should fail if there's a precompile in the address. If there's no precompile, the call will return without executing anything successfully.

I think I can make this more reliable by also passing zero gas in the call, then there is absolutely no way the call can succeed if there's a precompile.

> Also, need to check that precompiles before are also precompiles

That should be tested in a separate dedicated test that aims to test each specific EIP that adds the precompile. It's outside the scope of this PR.